File: js/packages/teams-ai/src/AzureOpenAIPlanner.ts
/**
 * @module teams-ai
 */
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

import { DefaultTurnState } from './DefaultTurnStateManager';
import { OpenAIClient, AzureOpenAIClient } from './OpenAIClients';
import { OpenAIPlanner, OpenAIPlannerOptions } from './OpenAIPlanner';
import { TurnState } from './TurnState';

/**
 * Additional options needed to use the Azure OpenAI service.
 */
export interface AzureOpenAIPlannerOptions extends OpenAIPlannerOptions {
    /**
     * The default model to use. This should be the model deployment name, not the model
     */
    defaultModel: string;
    /**
     * Endpoint for your Azure OpenAI deployment.
     */
    endpoint: string;

    /**
     * Optional. Which Azure API version to use. Defaults to latest.
     */
    apiVersion?: string;
}

/**
 * Planner that uses the Azure OpenAI service.
 * @template TState Optional. Type of the applications turn state.
 */
export class AzureOpenAIPlanner<TState extends TurnState = DefaultTurnState> extends OpenAIPlanner<
    TState,
    AzureOpenAIPlannerOptions
> {
    /**
     * @param {AzureOpenAIPlannerOptions} options Options for the AzureOpenAIPlanner
     * @private
     * @returns {AzureOpenAIClient} A planner that uses Azure OpenAI to generate plans
     */
    protected createClient(options: AzureOpenAIPlannerOptions): OpenAIClient {
        return new AzureOpenAIClient({
            apiKey: options.apiKey,
            apiVersion: options.apiVersion,
            endpoint: options.endpoint
        });
    }
}

File: dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/AI/Planner/AzureOpenAIPlanner.cs
using Microsoft.Bot.Builder.M365.AI.Prompt;
using Microsoft.Extensions.Logging;
using Microsoft.SemanticKernel.AI.ChatCompletion;
using Microsoft.SemanticKernel.AI.TextCompletion;
using Microsoft.SemanticKernel.Connectors.AI.OpenAI.ChatCompletion;
using Microsoft.SemanticKernel.Connectors.AI.OpenAI.TextCompletion;

namespace Microsoft.Bot.Builder.M365.AI.Planner
{
    /// <summary>
    /// Planner that uses the Azure OpenAI service.
    /// </summary>
    /// <typeparam name="TState">Type of the applications turn state</typeparam>
    public class AzureOpenAIPlanner<TState> : OpenAIPlanner<TState, AzureOpenAIPlannerOptions>
        where TState : TurnState
    {
        public AzureOpenAIPlanner(AzureOpenAIPlannerOptions options, PromptManager<TState> promptManager, ILogger logger) : base(options, promptManager, logger)
        {
        }

        private protected override ITextCompletion _CreateTextCompletionService(AzureOpenAIPlannerOptions options)
        {
            return new AzureTextCompletion(
                options.DefaultModel,
                options.Endpoint,
                options.ApiKey
            );
        }

        private protected override IChatCompletion _CreateChatCompletionService(AzureOpenAIPlannerOptions options)
        {
            return new AzureChatCompletion(
                options.DefaultModel,
                options.Endpoint,
                options.ApiKey
            );
        }
    }
}
